File: backend/app/schemas/booking.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.booking import BookingStatus, BookingType


class BookingCreate(BaseModel):
    type: BookingType
    title: str
    confirmation_number: str | None = None
    provider: str | None = None
    check_in: str | None = None
    check_out: str | None = None
    amount: float | None = None
    notes: str | None = None
    status: BookingStatus = BookingStatus.confirmed


class BookingUpdate(BaseModel):
    type: BookingType | None = None
    title: str | None = None
    confirmation_number: str | None = None
    provider: str | None = None
    check_in: str | None = None
    check_out: str | None = None
    amount: float | None = None
    notes: str | None = None
    status: BookingStatus | None = None


class BookingOut(BaseModel):
    id: uuid.UUID
    trip_id: uuid.UUID
    type: BookingType
    title: str
    confirmation_number: str | None
    provider: str | None
    check_in: str | None
    check_out: str | None
    amount: float | None
    notes: str | None
    status: BookingStatus
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/types/booking.ts
export type BookingType = string;
export type BookingStatus = "confirmed" | "pending" | "cancelled";

export interface Booking {
  id: string;
  trip_id: string;
  type: BookingType;
  title: string;
  confirmation_number: string | null;
  provider: string | null;
  check_in: string | null;
  check_out: string | null;
  amount: number | null;
  notes: string | null;
  status: BookingStatus;
  created_at: string;
}

export interface BookingCreate {
  type: BookingType;
  title: string;
  confirmation_number?: string;
  provider?: string;
  check_in?: string;
  check_out?: string;
  amount?: number;
  notes?: string;
  status?: BookingStatus;
}

export interface BookingUpdate extends Partial<BookingCreate> {}
